src/test/java/ClientTest.java

```diff
@@ -7,13 +8,14 @@
 public class ClientTest {
 
     @Test
-    public void echoMessagesSentThroughSocket() throws IOException {
-        IOHelper stdIO = new IOHelper("Hello\nbye");
+    public void writeMessagesToSocketFromTerminal() throws IOException {
+        IOHelper stdIO = new IOHelper("Hello");
         IOHelper socketIO = new IOHelper("");
 
         SocketStub socketStub = new SocketStub(socketIO.getIn(), socketIO.getOut());
         new Client(stdIO.getIn(), stdIO.getOut(), socketStub);
 
-        assertEquals( "Hello\n" + "bye\n", socketIO.getOutput());
+        assertEquals( "Hello\n", socketIO.getOutput());
```

Comment:
This works by chance, not design: it could be that your JUnit Thread runs faster than the thread within 'Client` and consequently nothing would be written to `socketIO`.